### ISSUE TITLE
Make Javascript Not Terrible

### ISSUE THREAD

**Issue by Bixkitts** [Bixkitts/relic-mercs-game#1]
My javascript is so terrible I had to use UML for
the first time in my life to understand what was going on in
my own code (it's in the project plan).
If somebody makes it sane, you're a real one.
It's not much code, and it's not overly complex,
I'm just not a JS programmer!
We generally want to stick to vanilla JS though as a challenge (that's part of the problem!)
so refrain from adding funni stuff unless absolutely justified...

**Comment by DanonymousCoder** [comment]
Hi, I would love to work on this issue. Can I have it assigned to me?

**Comment by Bixkitts** [comment]
This is the main thing preventing me from working on the project; I just don't have
much fun writing JS (I need to read up on it and study it more).
I had a frontend guy, but he has even more jobs than I do.

TL;DR:
------------
Having someone take a look would be radical, of course I'd assign it!

Details:
-------------
Besides _maybe_ having a fun game at the end, the priority of the project is
to see how well we can work with minimal dependencies.
So, before introducing any new libraries/frameworks; consider if we can realistically
write just the functionality we need ourselves (The backend is handwritten in C,
so consider that complicated protocols will break me).

As for the current state of the JS itself: it's disorganised but otherwise
I'm fairly happy with the actual implementation details.
I'm not a very experienced graphics programmer, but I think we did okay
with the low level WebGL stuff!
Authentication system might be horrible though; I've never written one before and
there is a bare-bones attempt at one lurking in the code that I made
without consulting my frontend guy.

I love to have all the bad stuff roasted and corrected, thanks!
